### PR TITLE
Update CS1935 compiler message text to remove System.Core.dll reference

### PR DESCRIPTION
**Pull request by kfechter**
Fixes dotnet/roslyn#37165

Changes the using reference portion of the CS1935 message from 'Are you missing a reference to 'System.Core.dll' or a using directive for 'System.Linq'?' to 'Are you missing a using directive for 'System.Linq'?' and removes the references in all the comments and associated tests.

### CODE REVIEW

**Comment by kfechter** [comment]
> Looks good, thanks!

No problem.  I do have one question. The non English localization files were updated automatically when the resource file was changed. As I am not fluent enough in the languages to ensure the localizations are correct, what is the best way to ensure that all of the localization files are updated/reviewed accordingly?

**Comment by AlekseyTs** [comment]
> As I am not fluent enough in the languages to ensure the localizations are correct, what is the best way to ensure that all of the localization files are updated/reviewed accordingly?

Don't worry about this, the files are automatically modified to make sure translators review the changes and make appropriate adjustments.

**Comment by 333fred** [comment]
> As I am not fluent enough in the languages to ensure the localizations are correct, what is the best way to ensure that all of the localization files are updated/reviewed accordingly?

If we were required to be fluent enough in all those languages to do translations, I think we would all lose our jobs 😅

**Comment by kfechter** [comment]
> > As I am not fluent enough in the languages to ensure the localizations are correct, what is the best way to ensure that all of the localization files are updated/reviewed accordingly?
> 
> If we were required to be fluent enough in all those languages to do translations, I think we would all lose our jobs 😅

haha, that's fair. Guess I really only needed to ask if there was anything specific I had to do in that regard.

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 1]
```diff
@@ -3299,7 +3299,7 @@ A catch() block after a catch (System.Exception e) block can catch non-CLS excep
     <value>Could not find an implementation of the query pattern for source type '{0}'.  '{1}' not found.  Consider explicitly specifying the type of the range variable '{2}'.</value>
   </data>
   <data name="ERR_QueryNoProviderStandard" xml:space="preserve">
-    <value>Could not find an implementation of the query pattern for source type '{0}'.  '{1}' not found.  Are you missing a reference to 'System.Core.dll' or a using directive for 'System.Linq'?</value>
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/CSharpResources.resx:3302]: >reference to 'System.Core.dll' or a  [](start = 125, length = 36)

Should we still mention references, but without any specific names? Something like: "Are you missing references to required assemblies or a using directive for 'System.Linq'?"

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/CSharp/Portable/CSharpResources.resx (hunk 1 above)
Comment (kfechter) [reply src/Compilers/CSharp/Portable/CSharpResources.resx:3302]: That's a fair point. 

I don't see any issues adding that information to the message. mentioning the reference assemblies but not specific names might make it more clear then just mentioning the using. I can make that change here in a bit.

Comment (kfechter) [reply src/Compilers/CSharp/Portable/CSharpResources.resx:3302]: Got a bit sidetracked with work, but I just added this commit. 
https://github.com/dotnet/roslyn/pull/45920/commits/7a96e074a1f4159054c7052ae7d8da082b83e3f5

it changes the message to  "Are you missing required assembly references or a using directive for 'System.Linq'?"

**Comment by AlekseyTs** [comment]
@kfechter Would it be fine if we squash commits in the process of merging this PR?

**Comment by kfechter** [comment]
> @kfechter Would it be fine if we squash commits in the process of merging this PR?

yeah, I don't have any issues with that.

**Comment by AlekseyTs** [comment]
@kfechter Thanks for the contribution.

**Pull request merged**